src/http/local_async_normal.hpp

```diff
@@ -433,63 +482,39 @@ struct http_async_connection<Tag, version_major, version_minor>
+      boost::system::error_code report_code = is_timedout_ ? boost::asio::error::timed_out : ec;
+      boost::system::system_error error(report_code);
+      this->source_promise.set_exception(std::make_exception_ptr(error));
+      this->destination_promise.set_exception(std::make_exception_ptr(error));
       switch (state) {
         case version:
-          this->version_promise.set_exception(boost::copy_exception(error));
+          this->version_promise.set_exception(std::make_exception_ptr(error));
         case status:
-          this->status_promise.set_exception(boost::copy_exception(error));
+          this->status_promise.set_exception(std::make_exception_ptr(error));
         case status_message:
           this->status_message_promise.set_exception(
-              boost::copy_exception(error));
+              std::make_exception_ptr(error));
         case headers:
-          this->headers_promise.set_exception(boost::copy_exception(error));
+          this->headers_promise.set_exception(std::make_exception_ptr(error));
         case body:
           if (!callback) {
-            // N.B. if callback is non-null, then body_promise has
-            // already been set to value "" to indicate body is
-            // handled by streaming handler so no exception should be set
-            this->body_promise.set_exception(boost::copy_exception(error));
+            // N.B. if callback is non-null, then body_promise has already been
+            // set to value "" to indicate body is handled by streaming handler
+            // so no exception should be set
+            this->body_promise.set_exception(std::make_exception_ptr(error));
           }
+          else
+            callback( boost::iterator_range<typename std::array<typename char_<Tag>::type, 1024>::const_iterator>(), report_code );
           break;
         default:
           BOOST_ASSERT(false && "Bug, report this to the developers!");
       }
     }
   }
 
-  string_type parse_chunk_encoding(string_type& body_string) {
-    string_type body;
-    string_type crlf = "\r\n";
-
-    typename string_type::iterator begin = body_string.begin();
-    for (typename string_type::iterator iter =
-             std::search(begin, body_string.end(), crlf.begin(), crlf.end());
-         iter != body_string.end();
-         iter =
-             std::search(begin, body_string.end(), crlf.begin(), crlf.end())) {
-      string_type line(begin, iter);
-      if (line.empty()) break;
-      std::stringstream stream(line);
-      int len;
-      stream >> std::hex >> len;
-      std::advance(iter, 2);
-      if (!len) break;
-      if (len <= body_string.end() - iter) {
-        body.insert(body.end(), iter, iter + len);
-        std::advance(iter, len + 2);
-      }
-      begin = iter;
-    }
-
-    return body;
-  }
-
-  int timeout_;
-  boost::asio::deadline_timer timer_;
+  int64_t timeout_;
+  bool remove_chunk_markers_;
```

Comment:
I'm not sure how much value this flag provides. Can you expand on why you're using it, instead of just leaving the conditional checks as `if (this->is_chunk_encoding) {` ?

Reply:
And here.